packages/server/src/api/controllers/datasource.ts

```diff
@@ -118,24 +124,32 @@ async function invalidateVariables(
   await invalidateDynamicVariables(toInvalidate)
 }
 
-export async function update(ctx: UserCtx<any, UpdateDatasourceResponse>) {
+export async function update(
+  ctx: UserCtx<UpdateDatasourceRequest, UpdateDatasourceResponse>
+) {
   const db = context.getAppDB()
   const datasourceId = ctx.params.datasourceId
   const baseDatasource = await sdk.datasources.get(datasourceId)
-  const auth = baseDatasource.config?.auth
   await invalidateVariables(baseDatasource, ctx.request.body)
 
   const isBudibaseSource =
     baseDatasource.type === dbCore.BUDIBASE_DATASOURCE_TYPE
 
-  const dataSourceBody = isBudibaseSource
-    ? { name: ctx.request.body?.name }
+  const dataSourceBody: Datasource = isBudibaseSource
+    ? {
+        name: ctx.request.body?.name,
```

Comment:
This is the only major logical change, everything else is typing. These types should normally accompany a Budibase datasource type, unsure why they weren't here originally.